docs/testplan/Smartswitch-test-plan.md

```diff
@@ -0,0 +1,876 @@
+    - [1.9 Check the State of DPUs](#19-check-the-state-of-dpus)
+    - [1.10 Check the Health of DPUs](#110-check-the-health-of-dpus)
+    - [1.11 Check reboot cause history](#111-check-reboot-cause-history)
+    - [1.12 Check the DPU state after OS reboot](#112-check-the-dpu-state-after-os-reboot)
+
+## Introduction
+
+The purpose is to test the functionality of Smartswitch.
+Smartswitch is connected to dpu sleds via pcie link having two dpus per sled.
+
+## Scope
+
+The test is targeting a running SONIC on Switch and SONIC-DASH system on each dpus. 
+Purpose of the test is to verify smartswich platform related functionalities/features for each dpus. 
+For every test cases, all DPUs need to be powered on unless specified in any of the case.
+
+## Definitions and Abbreviations
+
+| **Term**   | **Meaning**                              |
+| ---------- | ---------------------------------------- |
+| DPU       | Data Processing Unit       |
+| NPU       | Network Processing Unit       |
+
+
+## Objectives of Test Cases
+
+|    | **Test Case**   | **Intention**                              |
+| ---------- | ---------- | ---------------------------------------- |
+| 1.1 | Check platform inventory       | To verify the DPU inventories shown in the cli |
+| 1.2 | Check platform voltage       |  To verify the Voltage sensor values and and functionality of alarm by changing the threshold values |
+| 1.3 | Check platform temperature       |  To Verify the Temperature sensor values and functionality of alarm by changing the threshold values |
+| 1.4 | Check dpu console       | To Verify console access for all DPUs       |
+| 1.5 | Check midplane ip address between NPU and DPU      | To Verify PCIe interface created between NPU and DPU according to bus number |
+| 1.6 | Check DPU shutdown and power up individually      |  To Verify one DPU shutdown  and other dpus in same as well in other sleds are up |
+| 1.7 | Check removal of pcie link between npu and dpu       | To Verify the PCie hot plug functinality        |
+| 1.8 | Check the NTP date and timezone between DPU and NPU       | To Verify NPU and DPU are in sync with respect to timezone and logs timestamp |
+| 1.9 | Check the State of DPUs      | To Verify DPU state details during online and offline      |
+| 1.10 | Check the Health of DPUs       | To Verify overall health (LED, process, docker, services and hw) of DPU |
+| 1.11 | Check reboot cause history       | To Verify reboot cause history cli |
+| 1.12 | Check the DPU state after OS reboot       | To Verify DPU state on host reboot |
+
+
+## Test Cases
+
+
+### 1.1 Check Platform Inventory
+
+#### Steps
+ * Use command `show platform inventory` to get inventory
+ * Get the number of dpu modules from PMON APIs - get_num_modules()
+
+#### Verify in
+ * Switch
+   
+#### Sample Output
+```
+On Switch:
+
+root@sonic:/home/cisco# show platform inventory 
+    Name                Product ID      Version              Serial Number   Description
```

Comment:
What does the version column mean in this table? Where is the data for this column taken from? We don't have an API defined for this.

Reply:
This is the standard column in show platform inventory and there is no change to it.

Reply:
The `show platform inventory` is not a standard command. It new command for the SONiC community that was introduced in the scope PMON HDL.

Reply:
Changed the test case to check the DPU state.

Reply:
Changed the command to check dpu status.